libraries/shared/logs/extractor.go

```diff
@@ -153,21 +154,52 @@ func (extractor LogExtractor) BackFillLogs(endingBlock int64) error {
 		return fmt.Errorf("error extracting logs: %w", ErrNoWatchedAddresses)
 	}
 
-	headers, headersErr := extractor.HeaderRepository.GetHeadersInRange(*extractor.StartingBlock, endingBlock)
-	if headersErr != nil {
-		logrus.Errorf("error fetching missing headers: %s", headersErr)
-		return fmt.Errorf("error getting unchecked headers to check for logs: %w", headersErr)
+	ranges, chunkErr := ChunkRanges(*extractor.StartingBlock, endingBlock, HeaderChunkSize)
+	if chunkErr != nil {
+		return fmt.Errorf("error chunking headers to lookup in logs backfill: %w", chunkErr)
 	}
 
-	for _, header := range headers {
-		err := extractor.fetchAndPersistLogsForHeader(header)
-		if err != nil {
-			return fmt.Errorf("error fetching and persisting logs for header with id %d: %w", header.Id, err)
+	for _, r := range ranges {
+		headers, headersErr := extractor.HeaderRepository.GetHeadersInRange(r[0], r[1])
+		if headersErr != nil {
+			logrus.Errorf("error fetching missing headers: %s", headersErr)
+			return fmt.Errorf("error getting unchecked headers to check for logs: %w", headersErr)
+		}
+
+		for _, header := range headers {
+			err := extractor.fetchAndPersistLogsForHeader(header)
+			if err != nil {
+				return fmt.Errorf("error fetching and persisting logs for header with id %d: %w", header.Id, err)
+			}
 		}
 	}
+
 	return nil
 }
 
+func ChunkRanges(startingBlock, endingBlock, interval int64) ([][]int64, error) {
```

Comment:
Opted to make this function public purely so that I could unit test it specifically - happy to make it private if that seems excessive at this point (since it's tested through `BackFillLogs` now)